File: CPP-Programing/DSA/array.cpp
#include<iostream>

class Array{
    private:
      int capacity;
      int lastindex;
      int *ptr;
    public:
      Array(int);
      void append(int);
      bool isFull();
      bool isEmpty();
      int count();
      int get(int);
};
// constructor body declaration
Array::Array(int cap){
    capacity = cap;
    lastindex = -1;
    ptr = new int[capacity];
}
// function declaration to add element data in the array class
void Array::append(int data){
    if(lastindex==capacity-1){
        std::cout<<"Array is Full"<<std::endl;
    }
    lastindex++;
    ptr[lastindex] = data;
}
// code for checking isFull;
bool Array::isFull(){
    return lastindex == capacity-1;
}
// code for checking isEmpty
bool Array::isEmpty(){
    return lastindex == -1;
}
// function for count total number of element in arrays
int Array::count(){
    return lastindex+1;
}
// function for return element at an index
int Array::get(int index){
    if(index>=lastindex && index<=capacity-1)
       return ptr[index];
}

File: Javascript/DSA/array.js
class Arrays{
    // this is variable in class
    capacity;
    lastindex;
    ptr;
    // creating a constructor to initialize class variable
    constructor(cap){
        this.capacity = cap;
        this.lastindex = -1;
        this.ptr = [];
    }
    // function to check array is empty 
    isEmpty(){
        return this.lastindex == -1;
    }
    // function is to check array is full
    isFull(){
        return this.lastindex==this.capacity-1;
    }
    // function is used to add data in the array
    append(data){
        if(this.isFull()){
            console.log("Arrays is Full");
        }
        else{
            this.lastindex++;
            this.ptr[this.lastindex]=data;
        }
    }
    // function is used to access the data at the index
    get(index){
        if(index>=0 && index<=this.lastindex)
           return this.ptr[index];
    }
    // function is used to count total element inside an arrays
    count(){
        return this.lastindex+1;
    }
    // function is used to insert a data at a specficed index
    insert(index,data){
        if(index>=this.lastindex || index<=this.capacity-1)
           this.ptr[index]=data;
    }
}
const  a = new Arrays(4);
a.append(56);
a.append(35);
a.append(90);
a.append(39);
console.log(a.isEmpty());
console.log(a.isFull());
console.log(a.count());
for(let i = 0;i<a.count();i++){
    console.log(a.get(i)+ " ");
}
a.insert(2,500);
console.log("After inserting data at a specific index");
for(let i=0;i<a.count();i++){
    console.log(a.get(i)+" ");
}
